app/Http/Controllers/TransactionController.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\User;
+use App\Models\Transaction;
+use App\Services\AuthorizationService;
+use App\Services\NotificationService;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+
+class TransactionController extends Controller
+{
+    protected AuthorizationService $authorization;
+    protected NotificationService $notification;
+    public function __construct(AuthorizationService $authorization, NotificationService $notification)
```

Comment:
```suggestion
    public function __construct(
        private AuthorizationService $authorization, 
        private NotificationService $notification
     ) {
     }
```

E pode retirar as propriedades da linha 14 e 15